istioctl/pkg/ztunnelconfig/ztunnelconfig.go

```diff
@@ -0,0 +1,486 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ztunnelconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"istio.io/istio/istioctl/pkg/cli"
+	"istio.io/istio/istioctl/pkg/completion"
+	"istio.io/istio/istioctl/pkg/kubeinject"
+	ambientutil "istio.io/istio/istioctl/pkg/util/ambient"
+	ztunnelDump "istio.io/istio/istioctl/pkg/writer/ztunnel/configdump"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	jsonOutput    = "json"
+	yamlOutput    = "yaml"
+	summaryOutput = "short"
+
+	defaultProxyAdminPort = 15000
+)
+
+var (
+	workloadsNamespace string
+	verboseProxyConfig bool
+
+	address, node string
+
+	// output format (json, yaml or short)
+	outputFormat string
+
+	proxyAdminPort int
+
+	configDumpFile string
+
+	labelSelector = ""
+)
+
+func ZtunnelConfig(ctx cli.Context) *cobra.Command {
+	configCmd := &cobra.Command{
+		Use:   "ztunnel-config",
+		Short: "Retrieve information about ztunnel configuration.",
+		Long:  "A group of commands used to retrieve information about Ztunnel configuration from a Ztunnel instance.",
+		Example: `  # Retrieve summary about workload configuration for a given ztunnel.
+	istioctl ztunnel-config <workload> <ztunnel-name[.namespace]>`,
+		Aliases: []string{"zc"},
+	}
+
+	configCmd.PersistentFlags().StringVarP(&outputFormat, "output", "o", summaryOutput, "Output format: one of json|yaml|short")
+	configCmd.PersistentFlags().IntVar(&proxyAdminPort, "proxy-admin-port", defaultProxyAdminPort, "Envoy proxy admin port")
+
+	configCmd.AddCommand(logCmd(ctx))
+	configCmd.AddCommand(workloadConfigCmd(ctx))
+
+	return configCmd
+}
+
+func workloadConfigCmd(ctx cli.Context) *cobra.Command {
+	var podName, podNamespace string
+
+	workloadConfigCmd := &cobra.Command{
+		Use:   "workload [<type>/]<name>[.<namespace>]",
+		Short: "Retrieves workload configuration for the specified ztunnel pod",
+		Long:  `Retrieve information about workload configuration for the ztunnel instance.`,
+		Example: `  # Retrieve summary about workload configuration for a given ztunnel.
+  istioctl proxy-config workload <ztunnel-name[.namespace]>
+
+  # Retrieve summary of workloads on node XXXX for a given ztunnel instance.
+  istioctl proxy-config workload <ztunnel-name[.namespace]> --node ambient-worker
+
+  # Retrieve full workload dump of workloads with address XXXX for a given ztunnel
+  istioctl proxy-config workload <ztunnel-name[.namespace]> --address 0.0.0.0 -o json
```

Comment:
This is a note for a future enhancement, not a review comment, but:

1. `node` is rather meaningless here as a filter.

1. It would actually be good to let you specify  `--node` and have this command auto-select the correct ztunnel for that node, rather than having to go find the right ztunnel for the node you care about yourself.

Reply:
+1 to both of these

Reply:
https://github.com/istio/istio/issues/50362 good ideas, cc/ @bleggett